---
title       : V2 and V3 antibodies as correlate of protection in RV144
subtitle    : VIDD Seminar Series, May 21, 2013
author      : Raphael Gottardo
job         : Associate Member, Vaccine and Infectious Diseases Division
framework   : io2012        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : tomorrow      # 
widgets     : []            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}
logo: Logo.png

--- &twocol w1:65% w2:33%

```{r knitr-options, echo = FALSE, eval=TRUE}
library(knitr)
opts_knit$set(progress = TRUE, verbose = TRUE)
opts_chunk$set(cache=TRUE, echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE, error=TRUE, echo=FALSE, dev="png", dev.args=list(bg="transparent"), out.width="900px",fig.width=12, fig.height=5, dpi=150, fig.align="center",autodep=TRUE)
options(width=80)
```

```{r generate-database, eval=FALSE}
# This is just given as an example, to show how the databases are generated.
library(filehash)
library(pepStat)
library(Pviz)
library(HIV.db)
library(PEP.db)
data(pep_hxb2)
# filehashOption(defaultType = "RDS")
# setwd("~/Dropbox/Work/PeptideArray/Data/HVTN204/")
# ## Initialize database
# db <- dbInit("./DataDB-new-quad2/")
# 
# ## Create a petptide set
# db$pSet <- makePeptideSet(path="./gprFiles/", mapping.file="./mapping.csv", log=TRUE, norm.empty=FALSE, rm.control.list=c("empty"), check.row.order=TRUE)
# ## Summarize peptide replicates by median
# db$psSet <- summarizePeptides(db$pSet, summary="median", position=pep_hxb2)
# ## Normalize data
# db$pnSet <- NormalizeArray(db$psSet, robust=TRUE, standard=FALSE, method="ZpepQuad")
# ## Smooth data using a running mean with widow 9AAs
# db$pnmSet <- slidingMean(db$pnSet, width=9)
# ## Compute frequency of responders
# db$freq <- makeCalls(db$pnmSet, cutoff=1.1, group=NULL, method="absolute")
# # Compute major hotspots (above 20% responders)
# db$hotspots <- .reduce2hotspots(db$pnmSet, db$freq, 20)
# ## Compute frequency of responders broken down by ad5 status
# db$freq.group <- makeCalls(db$pnmSet, cutoff=1.1, group="ad5", method="absolute")
# 
# ## Do the same as above but clade-specific
# clade.ordered<-pepStat:::clade(db$pnSet)[,c("M","A","B","C","D","CRF01","CRF02")]
# db$pnSet.clade <- split(db$pnSet, clade.ordered)
# db$pnmSet.clade <- lapply(db$pnSet.clade, slidingMean, width=9)
# db$freq.clade <- lapply(db$pnmSet.clade, makeCalls, cutoff=1.1, method="absolute", group=NULL)
# db$freq.group.clade <- lapply(db$pnmSet.clade, makeCalls, cutoff=1.1, method="absolute", group="ad5")
```

```{r set-up-database}
library(filehash)
data(pep_hxb2)
filehashOption(defaultType = "RDS")

### These are pre-computed and stored in a filehash database
filehashOption(defaultType = "RDS")

RV144<-dbInit("~/Dropbox/Work/PeptideArray/Data/rv144cc/DataDB-new-quad2/")
Vax003<-dbInit("~/Dropbox/Work/PeptideArray/Data/Vax003/DataDB-new-quad2/")
Vax004<-dbInit("~/Dropbox/Work/PeptideArray/Data/Vax004/DataDB-new-quad2/")
NSDP<-dbInit("~/Dropbox/Work/PeptideArray/Data/NSDP/DataDB-new-quad2/")
HVTN204<-dbInit("~/Dropbox/Work/PeptideArray/Data/HVTN204/DataDB-new-quad2/")
```

```{r load-up-packages}
library(pepStat)
library(Pviz)
library(HIV.db)
library(PEP.db)
library(RColorBrewer)
library(ggplot2)
library(plyr)
library(reshape2)
library(Biostrings)
```

```{r set-up-annotations}
## Color options for tracks
colorAnno <- brewer.pal(4, "Greys")
colorTrack <- brewer.pal(7, "Dark2")

## Colors from our css theme
grey<-c("#e6e6e6", "#a9a9a9", "#797979", "#515151")
red<-c("#BC1C5F","#CB457D","#DA729D")
yellow<-c("#ECEC0D","#F0F040","#F4F474")
green<-c("#4BB413","#6CC53C","#91D56C")
blue<-c("#2266AF","#4983C1","#74A2D3")
  
col.grid<-grey[4]
col.axis<-grey[2]
col.title<-grey[1]
col.range<-red[1]

## ggplot2 black theme
my_theme<-theme(rect=element_rect(color="black"),text=element_text(color=grey[1],size=20),axis.text=element_text(color=grey[1]),legend.key.size=unit(.8,"cm"),panel.margin = unit(.8, "lines"), legend.background=element_rect(fill="black"), panel.background=element_rect(fill="black"), plot.background=element_rect(color="black", fill="black"), panel.grid.minor=element_line(color=grey[4]), panel.grid.major=element_line(color=grey[3]), legend.key=element_rect(color="black", fill="black"), strip.background=element_rect("black"), strip.text=element_text(color=grey[1],size=20))

genome<-"hxb2"
## Load annotations
anno_db <- loadFeatures(genome=genome)

# Create protein tracks
proteins <- getFeature(anno_db, category="protein")
pro_anno <- AnnotationTrack(start=start(proteins), end=end(proteins), id=proteins$name, genome=genome,name="", chromosome="chr", fill=colorAnno[3:4], size=3, fontcolor="white", fontsize=14)
displayPars(pro_anno) <- list(alpha=1, showFeatureId=TRUE, rotation=0, background.title="black", fontcolor="white", stacking = "dense")

# Create track with landmarks
landmarks <- getFeature(anno_db, name=c("V1","V2","V3","V4","V5","ID","MPER","TM"))
start<-sort(c(start(landmarks),535,628))+1
end<-sort(c(end(landmarks),581,659))
id<-c("V1","V2","V3","V4","V5","HR1","ID","HR2","MPER","TM")
landmarks_anno <- AnnotationTrack(start=start, end=end, id=id, name="",chromosome="chr", fill=colorAnno[4], fontsize=14, size=3)
displayPars(landmarks_anno) <- list(alpha=1, showFeatureId=TRUE, rotation=0, background.title="black", shape = "ellipse")
```

## RV144 efficacy trial

*** left

<iframe src="http://www.nejm.org/doi/full/10.1056/NEJMoa0908492#t=abstract"></iframe>

*** right

* Prime-boost combination: ALVAC and AIDSVAX
* 31% efficacy
* How did the vaccine might work?


--- &twocol w1:65% w2:33%

## RV144 case-control study

*** left

<iframe src="http://www.nejm.org/doi/full/10.1056/NEJMoa1113425"></iframe>

*** right

* Pilot studies to downselect immune variables
* Six variables used in primary analysis
* Remaining variables used in secondary analyses
* Do these variables predict HIV-1 infection?

---

## Outline

* Peptide microarray overview
* RV144 immunogenicity
* Definition of our peptide microarray immune variables
* Review of the RV144 correlate study results
* Correlate analysis using our derived variables (focus on V2 and V3)
* Comparison of RV144 to other vaccine regiments

---

## Array design overview and goals

* Identify regions of the HIV viral envelope that elicit antibodies during natural infection or after vaccination
* Array of overlapping gp160 peptides:
    * Peptides mosaic designed by Bette Korber at LANL
    * 1423 peptides (15 mers) across 7 sub-types
    * Serum + secondary antibody = Reactive peptides

* Data analyzed as described in Imholte *et al.* (2013, submitted to *JIM*).

---

## Aggregate-response<sup>*</sup> rate vs. HxB2

```{r rv144-aggregate}
hotspots.highlight<-IRanges(start = RV144$hotspots$start, end=RV144$hotspots$end)

HXB2_axis <- ProteinAxisTrack(hotspots.highlight,fontsize=16, littleTicks=TRUE, addNC=TRUE, col=col.axis, fontcolor=col.title, lwd=4, col.range=col.range)

# Order the positions
ord<-order(pepStat:::position(RV144$pnSet))
pos<-pepStat:::position(RV144$pnSet)[ord]

RV144.freq.track <- DataTrack(data = RV144$freq[ord], start=pos, width=0, name = "Frequency of responders", chromosome="chr", type=c("g","l"),v=-20, size=20, genome="HxB2", col.axis=col.title)
displayPars(RV144.freq.track)<-list(col=colorTrack[1], lwd=4)

# Use a rotation of 45 degrees to visualize the gp41 annotations
displayPars(landmarks_anno)<-list(rotation=45)

plotTracks(c(pro_anno, landmarks_anno, RV144.freq.track, HXB2_axis),legend=TRUE,col.grid=colorAnno[4],cex.legend=1,fontsize.title=16,cex.axis=1.2,fontsize.legend=16, ylim=c(0,100),col.title=col.title,background.title="black",col.border.title="transparent",col.grid=col.grid)
```

<span class="source"><sup>*</sup>The aggregate response represents an average of all sub-types.</span> 
We can see 4 hotspots noted as <span class="red">C1, V2, V3, and C5.</span>

---

## Subtype-specific response rates

```{r rv144-subtype}
displayPars(landmarks_anno)<-list(size=3)

data.list<-lapply(1:7,function(i,x,y,dp,size,color,names,...){ord<-order(pepStat:::position(y[[i]]));pos<-pepStat:::position(y[[i]])[ord];DataTrack(data = x[[i]][ord], start=pos, width=0, name =names[i], chromosome="chr",type=c("g","l"),v=-20,h=-5,col=color[i],lwd=4,genome="HxB2")},x=RV144$freq.clade,y=RV144$pnSet.clade, size=sizeTrack, color = colorTrack,ylim=c(0,100),names=c("M","A","B","C","D","CRF01","CRF02"))

plotTracks(c(landmarks_anno,data.list,HXB2_axis),legend=TRUE,col.grid="grey90",cex.legend=1,fontsize.title=14,cex.axis=1.2,fontsize.legend=16,ylim=c(0,100),col.title=col.title,col.axis=col.title,background.title="black",col.border.title="transparent",col.grid=col.grid)
```

* <span class="white">V2</span>: largest response in CRF01 but <span class="red">virtually no B reactivity.</span>
* <span class="white">V3</span>: largest response in B, less so in CRF01. 

---

## Peptide variable definition

* Defined four reactive regions (hotspots): <span class="red">C1, V2, V3, and C5</span>
* For each region, find the position at the maximum response (*peak summit*)
    * <span class="white">**Aggregate response**</span>: average of all subtypes centered at the maximum position
    * <span class="white">**Subtype-specific responses**</span>: corresponding individual peptide responses
* Total of <span class="white">24 variables</span>: 4 aggregate and 20 subtype variables
* Correlate analysis for these 24 variables using two phase design logistic regression (Haynes *et al.*, 2012)

---

## RV144 case-control study overview
```{r set-primary-variables}
# Source nessecary code (Holly's)
source("~/Dropbox/Work/PeptideArray/Projects/PeptidePaper/lib/my.tps.function.R")
source("~/Dropbox/Work/PeptideArray/Projects/PeptidePaper/lib/primaryinferences_cch.r")
source("~/Dropbox/Work/PeptideArray/Projects/PeptidePaper/lib/primaryinferences_main_090111.r")

datPrim<-read.csv("~/Dropbox/Work/PeptideArray/Data/rv144cc/rv144_data_wk26_correlates_primary_scaled_single_imp.csv",header=T)
datSec<-read.csv("~/Dropbox/Work/PeptideArray/Data/rv144cc/rv144_data_wk26_correlates_secondary_scaled.csv",header=T)

pd<-pData(RV144$pnSet)
pd<-subset(pd,!duplicated(ptid))

v2_scaffold<-read.csv("~/Dropbox/git/Talks/VIDD-0513/data/tomaras_v2_scaffold_case_control.csv")
v2_scaffold<-v2_scaffold[order(v2_scaffold$pin),]
v2_scaffold<-subset(v2_scaffold, pin%in%pd$ptid)
#v2_scaffold<-ddply(v2_scaffold,.(outcome,pin),summarize,y=(y[visit==800]/y[visit==100]))
v2_92TH023<-subset(v2_scaffold,outcome=="gp70.AE.AP" & visit==800)
v2_A244<-subset(v2_scaffold,outcome=="tags.AE.A244.293F" & visit==800)

v2_scaffold<-read.csv("~/Dropbox/git/Talks/VIDD-0513/data/zolla_pazner_v2_scaffold_case_control.csv")
v2_scaffold<-v2_scaffold[order(v2_scaffold$pin),]
v2_scaffold<-subset(v2_scaffold, pin%in%pd$ptid)
v2_92TH023<-subset(v2_scaffold,outcome=="gp70.AE-V1V2.AP" & visit==800)

# Data.frame of primary variables
VP<-data.frame(CD4=CD4ICSprim,ADCC=ADCCprim,Avid=Avidprim,IgA=IgAprim,NAb=NAbprim,gp70V1V2=V2prim)

# Define the 4 aggregate hoptspot variables
ind<-pepStat:::position(RV144$pnSet)%in%RV144$hotspots$center & !duplicated(pepStat:::position(RV144$pnSet))
yNM1<-slidingMean(RV144$pnSet,width=2)

#Scale the data and define primary peptide variables
yNM1<-t(pepStat:::.bgCorrect.pSet(yNM1))
VPP<-as.data.frame(scale(yNM1[,ind]))
VPP.notscaled<-as.data.frame(yNM1[,ind])
colnames(VPP)<-c("C1","V2","V3","C5")
colnames(VPP.notscaled)<-colnames(VPP)
y<-data.frame(VPP,VP)

# Individual peptide responses
yN<-t(pepStat:::.bgCorrect.pSet(RV144$pnSet))
ind<-pepStat:::position(RV144$pnSet)%in%RV144$hotspots$center
VPS<-as.data.frame(scale(yN[,ind]))
VPS.notscaled<-as.data.frame(yN[,ind])
anno<-as.character(ranges(RV144$pnSet[ind,])$featureID)
lclade<-sapply(1:4,function(i){sum(pepStat:::position(RV144$pnSet)%in%(RV144$hotspots$center[i]))})
tmp<-unlist(sapply(1:length(lclade),function(i,x,y){rep(x[i],y[i])},x=c("C1","V2","V3","C5"),y=lclade))

colnames(VPS)<-paste(tmp,".",unlist(lapply(anno,function(x){strsplit(x,"_")}[[1]][2])),sep="")

colnames(VPS.notscaled)<-colnames(VPS)

# 20 columns for VPS added to the dataframe
# Global data frame
y<-cbind(y,VPS)
```

```{r table-correlate}
res<-tps.univariate.matrix(as.matrix(y))
res<-as.data.frame(cbind(t(res),p.adjust(res[4,],method="BH")))
names(res)<-c("OR","CI.low","CI.up","pvalue","q-value")
res$variable<-rownames(res)
res<-within(res,aggregate<-variable)
res<-within(res,type<-variable)
res$aggregate[1:4]<-"Yes"
res$aggregate[-c(1:4)]<-"No"
res$type[5:10]<-"Primary"
res$type[-c(5:10)]<-"Peptide"

## What if we set to zero values below the positivity threshold?
# VPS2<-VPS.noscaled
# VPS2[VPS2<1.1]<-0
# res<-tps.univariate.matrix(as.matrix(VPS2))
# res<-as.data.frame(cbind(t(res),p.adjust(res[4,],method="BH")))
# names(res)<-c("OR","CI.low","CI.up","pvalue","q-value")
```

* <span class="white"> **Six primary variables** </span>: 5 antibody and 1 cellular responses
    * Plasma IgA antibodies (IgA); The avidity of IgG antibodies; Antibody-dependent cellular cytotoxicity (ADCC); HIV-1 neutralizing antibodies; The binding of IgG antibodies to variable regions 1 and 2 (V1V2) of the gp120 Env (gp70V1V2); The level of Env-specific CD4+ T cells
* <span class="white"> **Results<sup>*: </sup>**</span>
    * gp70V1V2 inversely correlated with infection (estimated odds ratio, `r res["gp70V1V2","OR"]` per 1-SD increase; p=`r res["gp70V1V2","pvalue"]`)
    * IgA correlated with infection (estimated odds ratio, `r res["IgA","OR"]` per 1-SD increase; p=`r res["IgA","pvalue"]`)

<span class="source"><sup> * </sup>Based on univariate model.</span> 

---

## Odds-ratios for aggregate and primary variables

```{r OR-ggplot-aggregate}
res2<-subset(res, aggregate=="Yes" | type=="Primary")
res2$variable<-as.factor(res2$variable)
res2$variable<-reorder(res2$variable, -res2$OR)

ggplot(res2,aes(x=variable))+geom_hline(yintercept = 1,color=grey[3],size=2)+geom_segment(aes(y=CI.low, yend=CI.up, x=variable, xend=variable, size=type), lineend = "round", color=grey[1])+geom_point(aes(y=OR,color=pvalue),size=6)+coord_flip()+scale_color_gradient( trans = 'log', low=yellow[1], high=red[1], breaks=c(0.05,0.15,0.5))+labs(x="",y="Odds Ratio")+scale_size_discrete(range=c(3,1))+scale_y_continuous(breaks=seq(0.4,2.0,.2))+my_theme


```

Most significant (inverse) correlation with our V2 aggregate variable. 

---

## Odds-ratios for V2 variables

```{r OR-ggplot-V2-subtype, fig.height=4.5}
res2<-subset(res, grepl("V2",variable))
res2$variable<-as.factor(res2$variable)
res2$variable<-reorder(res2$variable, -res2$OR)
ggplot(res2,aes(x=variable))+geom_hline(yintercept = 1,color=grey[3],size=2)+geom_segment(aes(y=CI.low, yend=CI.up, x=variable, xend=variable, size=type), lineend = "round", color=grey[1])+geom_point(aes(y=OR,color=pvalue),size=6)+coord_flip()+scale_color_gradient( trans = 'log', low=yellow[1], high=red[1], breaks=c(0.05,0.15,0.5))+labs(x="",y="Odds Ratio")+scale_size_discrete(range=c(3,1))+scale_y_continuous(breaks=seq(0.4,2.0,.2))+my_theme
```
* The V2 subtype-CRF01_AE (V2.1) peptide has the smallest estimated OR (OR=`r res2$OR[res2$variable=="V2.1"]`, p=`r res2$pvalue[res2$variable=="V2.1"]`). 
* No decrease in risk with the V2 subtype-B peptide.

---

## V2 tail distribution by subtypes

```{r tail-ggplot-V2-subtype, out.width="900px", fig.width=12, fig.height=6}
pd<-pData(RV144$pnSet)[!duplicated(pData(RV144$pnSet)$ptid),]
  pd$Risk <- factor(pd$BRA_risk, levels=unique(as.character(pd$BRA_risk)))
  pd$Infection <- pd$infec
  pd$Gender<-pd$dem_sex
  data<-cbind(VP, VPP.notscaled, VPS.notscaled, pd)
  
#  ind<-grep("V2.",names(data))
  ind<-grep("V2",names(data),value=TRUE)
  dt<-melt(data,measure.vars=ind)

  dt<-ddply(.data=dt,.variables=.(Infection,variable),transform,median=median(value))
  levels(dt$variable)[2]<-"V2.aggregate"

  new.data<-ddply(.data=dt,.variables=.(Infection,variable),transform,cdf=ecdf(value)(value))
  ggplot(new.data, aes(x=value))+geom_step(aes(x=value,y=1-cdf,col=Infection),size=1.5)+facet_wrap(~variable,scale="free_x",ncol=4)+xlim(c(-1,5))+scale_color_brewer(palette="Dark2")+scale_size_manual(values=c(1.5,2))+ylab("Prob(X>value)")+labs(x="Normalized intensity")+my_theme+theme(legend.position=c(0.9, 0.25),legend.key.size=unit(1.5,"cm"))

```

---

## Odds-ratios for C5 variables

```{r OR-ggplot-C5-subtype}
res2<-subset(res, grepl("C5",variable))
res2$variable<-as.factor(res2$variable)
res2$variable<-reorder(res2$variable, -res2$OR)
ggplot(res2,aes(x=variable))+geom_hline(yintercept = 1,color=grey[3],size=2)+geom_segment(aes(y=CI.low, yend=CI.up, x=variable, xend=variable, size=type), lineend = "round", color=grey[1])+geom_point(aes(y=OR,color=pvalue),size=6)+geom_point(aes(y=OR,color=pvalue),size=6)+coord_flip()+scale_color_gradient( trans = 'log', low=yellow[1], high=red[1], breaks=c(0.05,0.15,0.5))+labs(x="",y="Odds Ratio")+scale_size_discrete(range=c(3,1))+scale_y_continuous(breaks=seq(0.4,2.0,.2))+my_theme
```
The C5 subtype A peptide is significantly correlated with infection.

---

## C5 tail distribution by subtypes

```{r tail-ggplot-C5-subtype, out.width="900px", fig.width=12, fig.height=6}
  pd<-pData(RV144$pnSet)[!duplicated(pData(RV144$pnSet)$ptid),]
  pd$Risk <- factor(pd$BRA_risk, levels=unique(as.character(pd$BRA_risk)))
  pd$Infection <- pd$infec
  pd$Gender<-pd$dem_sex
  data<-cbind(VP, VPP.notscaled, VPS.notscaled, pd)
  
#  ind<-grep("V2.",names(data))
  ind<-grep("C5",names(data),value=TRUE)
  dt<-melt(data,measure.vars=ind)

  dt<-ddply(.data=dt,.variables=.(Infection,variable),transform,median=median(value))
  levels(dt$variable)[1]<-"C5.aggregate"

  new.data<-ddply(.data=dt,.variables=.(Infection,variable),transform,cdf=ecdf(value)(value))
  ggplot(new.data, aes(x=value))+geom_step(aes(x=value,y=1-cdf,col=Infection),size=1.5)+facet_wrap(~variable,scale="free_x",ncol=4)+xlim(c(-1,7))+scale_color_brewer(palette="Dark2")+scale_size_manual(values=c(1.5,2))+ylab("Prob(X>value)")+labs(x="Normalized intensity")+my_theme+theme(legend.position=c(0.9, 0.25),legend.key.size=unit(1.5,"cm"))
```

---

## How do gp70V1V2 and V2 variables compare?

```{r V2-gp70-V1V2, fig.height=4.5}
  pd<-pData(RV144$pnSet)[!duplicated(pData(RV144$pnSet)$ptid),]
  pd$Risk <- factor(pd$BRA_risk, levels=unique(as.character(pd$BRA_risk)))
  pd$Infection <- pd$infec
  pd$Gender<-pd$dem_sex
  data<-cbind(VP, VPP, VPS, gp70AEV1V2=v2_92TH023$y, pd)
  
  ind<-which(names(data)%in%c("V2","V2.1"))
  dt<-melt(data,measure.vars=ind)

  levels(dt$variable)[1]<-"V2.aggregate"

  df.cor <- ddply(dt, .(variable), function(myDF) {r = cor.test(myDF$value,myDF$gp70V1V2); data.frame(label=sprintf("r = %.2f", r$estimate, r$p.value),x=-2.6,y=2.5);});


  ggplot(dt, aes(x=value,y=gp70V1V2))+ geom_smooth(method = "lm",color=red[1],size=2) + geom_point(size=2,color=grey[1],alpha=.9)+facet_wrap(~variable,scale="free_x",ncol=3)+scale_color_brewer(palette="Dark2")+scale_size_manual(values=c(1.5,2))+labs(x="Normalized intensity")+my_theme+theme(legend.position=c(0.9, 0.25),legend.key.size=unit(1.5,"cm"))+xlim(c(-3,5))+geom_text(data=df.cor, aes(x=x, y=y, label=label), size=7, color=red[2])

# AE subtype below, results are similar
#   df.cor <- ddply(dt, .(variable), function(myDF) {r = cor.test(myDF$value,myDF$gp70AEV1V2); data.frame(label=sprintf("r = %.2f", r$estimate, r$p.value),x=-2.6,y=2.5);});
# 
# 
#   ggplot(dt, aes(x=value,y=gp70AEV1V2))+ geom_smooth(method = "lm",color=red[1]) + geom_point(size=2,color=grey[1])+facet_wrap(~variable,scale="free_x",ncol=3)+scale_color_brewer(palette="Dark2")+scale_size_manual(values=c(1.5,2))+labs(x="Normalized intensity")+my_theme+theme(legend.position=c(0.9, 0.25),legend.key.size=unit(1.5,"cm"))+xlim(c(-3,5))+geom_text(data=df.cor, aes(x=x, y=y, label=label), size=5, color="white")


```

* Weak correlation between the two V2 variables
* Could the two be measuring different antibodies? Likely, yes (Liao *et al.*, 2013).

--- &twocol w1:60% w2:38%

## V2 sieve analysis

*** left

<iframe src="http://www.nature.com/nature/journal/v490/n7420/full/nature11519.html"></iframe>

*** right

* V2 focused Analysis of 936 HIV-1 genome sequences from 44 vaccine and 66 placebo recipients
* Two genetic signatures of RV144 vaccination: K169 and L181


--- &twocol w1:45% w2:45%

## V2 binding and K169

*** left

* Lowest binding in V2 is in subtype B

```{r ggplot-V2-boxplot, out.width="480px", fig.width=6, fig.height=4}
  data<-melt(VPS.notscaled)
  data<-subset(data,grepl("V2",variable))
  ggplot(data, aes(x=variable,y=value))+geom_boxplot(aes(x=variable,fill=variable),col=grey[1], size=.5, outlier.colour=grey[1], outlier.size=3)+scale_fill_manual(values=colorTrack[1:5])+my_theme+ylim(c(-3,5))+theme(legend.position="none")+ylab("Normalized intensity")
```

*** right

* Subtype B be at position 169 is not a K

```{r K169, out.width="480px", fig.width=6, fig.height=4}
multiAlign<-readAAStringSet(system.file("extdata/alignments/Musclehxb2_vaccineInsert.fasta", package="PEP.db"))
path<-system.file("extdata/alignments/Musclehxb2_vaccineInsert.fasta", package="PEP.db")
res<-readAlign(path)

seq.hxb2<-ProteinSequenceTrack(multiAlign["HXB2"],name="HXB2",size=3,fontsize.title=14)

## Color for sequences
fcol<-rep(colorTrack[1],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.1<-ProteinSequenceTrack(as.character(multiAlign[[" CRF01"]]),name="1",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[2],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.A<-ProteinSequenceTrack(as.character(multiAlign[[" A"]]),name="A",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[3],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
fcol["V"] <- "white"
seq.B<-ProteinSequenceTrack(as.character(multiAlign[[" B"]]),name="B",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[4],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.D<-ProteinSequenceTrack(as.character(multiAlign[[" D"]]),name="D",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[5],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.M<-ProteinSequenceTrack(as.character(multiAlign[[" M"]]),name="MC2",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)


data(pep_hxb2)
pep_hxb2_ext<-coord2ext(pep_hxb2,refScale=res[[1]])

## Annotations track
## ProteinAxisTrack using the extended coordinates system
rpref <- ProteinAxisTrack(refScale=res[[1]],fontsize=18, littleTicks=TRUE, addNC=FALSE, col=col.axis, lwd=4, col.range=col.range, fontsize.title=16)

data.list2<-c(seq.1,seq.A,seq.B,seq.D,seq.M) 


plotTracks(trackList=c(data.list2,rpref), from=187, to=202, ylim=c(0,3),col.grid=colorAnno[4],cex.legend=1,cex.axis=1.2,fontsize.legend=16, ylim=c(0,100),col.title=col.title,background.title="black",col.border.title="transparent",col.grid=col.grid, fontsize=18, fontsize.title=16, col.title=grey[3])
```

---

```{r V3-interactions}
x<-sapply(1:20,function(i,y,VP){lapply(VP,function(x,y){round(tps.univariate.interaction(y,x),5)},y[,i])},y=VPS,VP=VP)
colnames(x)<-colnames(VPS)
```

## Interaction analyses

* Replicated the analysis presented in Haynes *et al.* (2012)

* Look for interaction between IgA and all peptide microarray variables

>- Significant interaction between IgA an most V3 variables
    * Interaction between IgA and V3-CRF01 (p= `r x["IgA","V3.1"]` )

---

## Effect of IgA on V3

```{r interaction-IgA-levels}
# Compute the ORs at different levels of IgA
table.V3.1<-tps.interaction.table(VPS$"V3.1",VP$IgA)
table.V3.1$variable<-"V3.1"
table.V3.1$levels<-c("20%","50%","80%")
# Compute the same for ADCC as a comparison 
table.ADCC<-tps.interaction.table(VP$ADCC,VP$IgA)
table.ADCC$variable<-"ADCC"
table.ADCC$levels<-c("20%","50%","80%")

data<-rbind(table.V3.1,table.ADCC)
names(data)[4]<-"pvalue"
data$levels<-reorder(data$levels,c(20,50,80,20,50,80))

ggplot(data,aes(x=levels))+geom_hline(yintercept = 1, color=grey[3],size=2)+geom_segment(aes(y=CI.low, yend=CI.up, x=levels, xend=levels), lineend = "round", color=grey[1],size=3)+geom_point(aes(y=OR,color=pvalue),size=6)+coord_flip()+scale_color_gradient( trans = 'log', low=yellow[1], high=red[1], breaks=c(0.005,0.05,0.5))+labs(x="",y="Odds Ratio")+scale_y_continuous(breaks=seq(0.4,2.0,.2))+my_theme+facet_grid(~variable)+xlab("IgA levels")
```


* Significant risk-reduction at low IgA level (p=`r data$pvalue[data$variable=="V3.1" & data$levels=="20%"]` at 20% and p=`r data$pvalue[data$variable=="V3.1" & data$levels=="50%"]` at 50%)

---

## Conditional V3 tail distributions

```{r condition-tail-ggplot-V3-subtype, out.width="900px", fig.width=12, fig.height=5.5}
data<-cbind(VP, VPS.notscaled, pd)

data<-within(data,Level.IgA<-as.factor(IgA> median(IgA)))
levels(data$Level.IgA)<-c("Low","High")

ind<-grep("Level.",names(data))
dt<-melt(data,measure.vars=ind)

names(dt)[names(dt)=="value"]<-"Level"
names(dt)[names(dt)=="V3.1"]<-"value"
dt<-ddply(.data=dt,.variables=.(Infection,variable,Level),transform,median=median(value))
new.data<-ddply(.data=dt,.variables=.(Infection,variable,Level),transform,cdf=ecdf(value)(value))

ggplot(new.data, aes(x=value))+geom_step(aes(x=value,y=1-cdf,col=Infection),size=1.5)+facet_wrap(~Level,ncol=4)+xlim(c(-1,6))+scale_color_brewer(palette="Dark2")+scale_size_manual(values=c(1.5,2))+ylab("Prob(X>value)")+labs(x="Normalized intensity")+my_theme

```
IgA level dichotomized as below (*Low*) or above (*High*) median(IgA)

--- &twocol w1:45% w2:45%

## V3 binding and subtype sequences

*** left

* Some variability in binding intensity

```{r ggplot-V3-boxplot, out.width="480px", fig.width=6, fig.height=4}
  data<-melt(VPS.notscaled)
  data<-subset(data,grepl("V3",variable))
  ggplot(data, aes(x=variable,y=value))+geom_boxplot(aes(x=variable,fill=variable),col=grey[1], size=.5, outlier.colour=grey[1], outlier.size=3)+scale_fill_manual(values=colorTrack)+my_theme+ylim(c(-2,6))+theme(legend.position="none")+ylab("Normalized intensity")
```

*** right

* V3 hotspot is not as well conserved as the V2 one

```{r V3-sequences, out.width="480px", fig.width=6, fig.height=4}
multiAlign<-readAAStringSet(system.file("extdata/alignments/Musclehxb2_vaccineInsert.fasta", package="PEP.db"))
path<-system.file("extdata/alignments/Musclehxb2_vaccineInsert.fasta", package="PEP.db")
res<-readAlign(path)

fcol<-rep(grey[3],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.hxb2<-ProteinSequenceTrack(multiAlign["HXB2"],name="HXB2",size=3,fontsize.title=14, fontface="bold", fontcolor=fcol)

## Color for sequences
fcol<-rep(colorTrack[1],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.1<-ProteinSequenceTrack(as.character(multiAlign[[" CRF01"]]),name="1",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[2],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.2<-ProteinSequenceTrack(as.character(multiAlign[[" CRF02"]]),name="2",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[3],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.A<-ProteinSequenceTrack(as.character(multiAlign[[" A"]]),name="A",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[4],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.B<-ProteinSequenceTrack(as.character(multiAlign[[" B"]]),name="B",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[5],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.C<-ProteinSequenceTrack(as.character(multiAlign[[" C"]]),name="C",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[6],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.D<-ProteinSequenceTrack(as.character(multiAlign[[" D"]]),name="D",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)
fcol<-rep(colorTrack[7],length(alphabet(multiAlign[[" M"]])))
names(fcol)<-alphabet(multiAlign[[" M"]])
seq.M<-ProteinSequenceTrack(as.character(multiAlign[[" M"]]),name="M",size=2,background.title="white",fontsize.title=14, fontface="bold", fontcolor=fcol)


data.list2<-c(seq.1,seq.2,seq.A,seq.B,seq.C,seq.D,seq.M) 

plotTracks(trackList=c(data.list2,rpref,seq.hxb2), from=327, to=344, ylim=c(0,3),col.grid=colorAnno[4],cex.legend=1,cex.axis=1.2,fontsize.legend=16, ylim=c(0,100),col.title=col.title,background.title="black",col.border.title="transparent",col.grid=col.grid, fontsize=18, fontsize.title=16, col.title=grey[3])
```

---

## Summary

* Our peptide microarray is a nice technology for profiling antibodies and mapping *linear* epitopes
* Validated the <del>V1</del>V2 correlate
    * Support the K169 sieve result
    * Non-neutralizing antibodies might be more important that we thought
* Interesting (conditional) V3 correlate
    * Relation to sieve results?
    * More work to be done

<aside class="note">
<section>
* V2 and not V1?
* Non conformational antibodies are also important
</section>
</aside>

---

## How does RV144 compare?

```{r comparison-to-other-vax}
displayPars(landmarks_anno)<-list(size=2)
## Reorder position for plotting
ord<-order(pepStat:::position(RV144$pnSet))
pos<-pepStat:::position(RV144$pnSet)[ord]
RV144.freq.track <- DataTrack(data = RV144$freq[ord], start=pos, width=0, name = "RV144", chromosome="chr",type=c("g","l"),genome="HxB2")
displayPars(RV144.freq.track)<-list(col=colorTrack[1],lwd=4,v=-20, col.axis=colorAnno[2])
Vax003.freq.track <- DataTrack(data = Vax003$freq[ord], start=pos, width=0, name = "Vax003", chromosome="chr",type=c("g","l"),genome="HxB2")
displayPars(Vax003.freq.track)<-list(col=colorTrack[2],lwd=4,v=-20,col.axis=colorAnno[2])
Vax004.freq.track <- DataTrack(data = Vax004$freq[ord], start=pos, width=0, name = "Vax004", chromosome="chr",type=c("g","l"),genome="HxB2")
displayPars(Vax004.freq.track)<-list(col=colorTrack[3],lwd=4,v=-20,col.axis=colorAnno[2])
NSDP.freq.track <- DataTrack(data = NSDP$freq[ord], start=pos, width=0, name = "HIV-1+", chromosome="chr",type=c("g","l"),v=-20,genome="HxB2")
displayPars(NSDP.freq.track)<-list(col=colorTrack[4],lwd=4,v=-20,col.axis=colorAnno[2])
HVTN.freq.track <- DataTrack(data = HVTN204$freq[ord], start=pos, width=0, name = "HVTN204", chromosome="chr",type=c("g","l"),v=-20,genome="HxB2")
displayPars(HVTN.freq.track)<-list(col=colorTrack[5],lwd=4,v=-20,col.axis=colorAnno[2])

plotTracks(c(landmarks_anno,RV144.freq.track,Vax003.freq.track,Vax004.freq.track,NSDP.freq.track, HVTN.freq.track, HXB2_axis),from=1,to=525,legend=TRUE,col.grid="grey90",cex.legend=1,fontsize.title=14,cex.axis=1.2,fontsize.legend=16, ylim=c(0,100),col.title=col.title,col.axis=col.title,col.border.title="transparent",background.title="black",col.grid=col.grid)
```
No <span class = 'red'>V2</span> in Vax004, HVTN204, HIV-1+<sup>*</sup><span class="source"><sup> * </sup>Depends on the virus subtype</span>. <span class = 'red'>V2</span> reactivity in Vax003.

--- .thank-you-slide

## Acknowledgments

<aside class="gdbar right"><img src="assets/img/Logo.png"></aside>

* David Montefiori, Xiaoying Shen and Georgia D. Tomaras (Duke)
* <span class="yellow">Greg Imholte, Mike Jiang, Renan Sauteraud</span>, Steve Self, Peter Gilbert, Paul Edlefsen (FHCRC)
* Robert Bailer and Ellen Turk (NIH/NVITAL)
* John Mascola (NIH)
* Bette Korber (LANL)
* MHRP, HVTN and CAVD.



<pre class="prettyprint", data-lang="URL">
<span class="gray4">Want more? &#8594; </span> <a href="http://raphg.github.io/Talks/VIDD-0513/index.html">http://goo.gl/57NhP</a>
</pre>

<a href="https://github.com/raphg/Talks/tree/gh-pages/VIDD-0513"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_white_ffffff.png" alt="Fork me on GitHub"></a>

